packages/flutter/lib/src/material/selectable_text.dart

```diff
@@ -30,7 +30,12 @@ class _TextSpanEditingController extends TextEditingController {
   final TextSpan _textSpan;
 
   @override
-  TextSpan buildTextSpan({required BuildContext context, TextStyle? style, required bool withComposing}) {
+  TextSpan buildTextSpan({
+    required BuildContext context,
```

Comment:
Breaking change. 2 breaking sites in g3.

Reply:
Yeah I think it's the same as the change that added `BuildContext` to this a few weeks ago. It is a bit sad, I could have gotten it from the `BuildContext` if it wasn't for `EditableText` has its own `textDirection` field. 

A downside of doing it in `buildTextSpan` is if someone overrides `buildTextSpan`, it isn't always easy to call the super implementation, as you may want to split the text into spans differently.

Reply:
Could we not require this, and simply have not providing the direction mean that the developer is opting out of the textspan appending action?

We can then just make sure all instances in flutter are opting-in